### PR TITLE
Remove usage of homebrew-go-resources.

### PR DESCRIPTION
**Pull request by petermattis**
Fix nits revealed by `brew audit --strict build/cockroach.rb`.

[Review on Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/3498)

### CODE REVIEW

File: build/cockroach.rb [hunk 1]
```diff
@@ -150,17 +149,28 @@ def install
...
     # linker error:
     #
     #   make -C src/github.com/cockroachdb/cockroach build SKIP_BOOTSTRAP=1
+    #
+    # Something truly bizarre is going on. The error is:
+    #
+    # # github.com/cockroachdb/c-snappy
+    # cannot load DWARF output from $WORK/github.com/cockroachdb/c-snappy/_obj//_cgo_.o: decoding dwarf section info at offset 0x0: too short
+    #
+    # But that error only happens when building with make. Using go
```
Comment (bdarnell) [inline build/cockroach.rb]: Can you run `env` inside the brew build to see what environment variables it's setting? In particular if it's setting `$LDFLAGS` our makefile will pick that up and pass it to go but a plain `go build` wouldn't use it.

Comment (petermattis) [inline build/cockroach.rb]: I've already looked and haven't seen anything suspicious. Perhaps you see something. Btw, this small makefile fails within the homebrew env:

```
foo:
        go build -x -v github.com/cockroachdb/c-snappy
```

```
GREP_COLOR=1;32
COLOR_NC=\e[0m
TERM_PROGRAM=iTerm.app
COLOR_RED=\e[0;31m
COLOR_CYAN=\e[0;36m
HOMEBREW_TEMP=/tmp
OBJCXX=clang++
OBJC=clang
COLOR_PURPLE=\e[0;35m
SHELL=/bin/bash
TERM=xterm-256color
MAKEFLAGS=-j8
HISTSIZE=1000000
CLICOLOR=1
HOMEBREW_CCCFG=sa
COLOR_LIGHT_RED=\e[1;31m
TMPDIR=/var/folders/qc/fpqpgdqd167c70dtc6840xxh0000gn/T/
GOBIN=/usr/local/Cellar/cockroach/0.1/bin
HOMEBREW_ERROR_PIPE=/tmp/homebrew20151218-16626-1faonm0/socket
DOCKER_HOST=tcp://192.168.99.100:2376
COLOR_LIGHT_PURPLE=\e[1;35m
COLOR_GREEN=\e[0;32m
Apple_PubSub_Socket_Render=/private/tmp/com.apple.launchd.xU0FpzNmpu/Render
GOHOME=/private/tmp/cockroach20151218-16645-rwk1e1
SDKROOT=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk
COLOR_BLUE=\e[0;34m
CMAKE_LIBRARY_PATH=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk/S
ystem/Library/Frameworks/OpenGL.framework/Versions/Current/Libraries
HOMEBREW_RUBY_PATH=/System/Library/Frameworks/Ruby.framework/Versions/Current/usr/bin/ruby
HOMEBREW_CXX=clang++
COLOR_YELLOW=\e[1;33m
GOROOT_BOOTSTRAP=/Users/pmattis/Development/go-1.4
USER=pmattis
HOMEBREW_OPTFLAGS=-march=native
DOCKER_MACHINE_NAME=default
CMAKE_FRAMEWORK_PATH=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk
/System/Library/Frameworks
SSH_AUTH_SOCK=/private/tmp/com.apple.launchd.wRzOOVn02y/Listeners
__CF_USER_TEXT_ENCODING=0x1F5:0x0:0x0
COLOR_BLACK=\e[0;30m
DOCKER_TLS_VERIFY=1
LSCOLORS=gxDxFxdxCxExExhbadgxgx
COLOR_GRAY=\e[1;30m
PATH=/usr/local/Library/ENV/4.3:/usr/local/opt/go/bin:/usr/local/opt/git/bin:/usr/bin:/bin:/usr/sbin:/sbin
HOMEBREW_ISYSTEM_PATHS=/usr/local/include:/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk/usr/include/libxml2:/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk/usr/include/apache2:/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk/System/Library/Frameworks/OpenGL.framework/Versions/Current/Headers
HOMEBREW_SDKROOT=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk
PWD=/private/tmp/cockroach20151218-16645-rwk1e1/src/github.com/cockroachdb/cockroach
HOMEBREW_BREW_FILE=/usr/local/bin/brew
COLOR_LIGHT_CYAN=\e[1;36m
EDITOR=emacs -nw
PKG_CONFIG_LIBDIR=/usr/lib/pkgconfig:/usr/local/Library/ENV/pkgconfig/10.11
HOMEBREW_ARCHFLAGS=
ITERM_PROFILE=Default
XPC_FLAGS=0x0
PS3=#?
CXX=clang++
HISTCONTROL=ignoreboth
XPC_SERVICE_NAME=0
COLOR_BROWN=\e[0;33m
DOCKER_CERT_PATH=/Users/pmattis/.docker/machine/machines/default
LESSCHARSET=utf-8
SHLVL=2
PS4=+
COLORFGBG=12;8
HOME=/private/tmp/cockroach20151218-16645-rwk1e1/.brew_home
HOMEBREW_PREFIX=/usr/local
CMAKE_INCLUDE_PATH=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk/usr/include/libxml2:/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk/usr/include/apache2:/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk/System/Library/Frameworks/OpenGL.framework/Versions/Current/Headers
COLOR_LIGHT_GRAY=\e[0;37m
ITERM_SESSION_ID=w0t0p1
MACOSX_DEPLOYMENT_TARGET=10.11
COLOR_LIGHT_BLUE=\e[1;34m
LOGNAME=pmattis
HOMEBREW_LIBRARY_PATHS=/usr/local/lib:/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk/System/Library/Frameworks/OpenGL.framework/Versions/Current/Libraries
HOMEBREW_PREFER_CLT_PROXIES=1
HOMEBREW_CC=clang
GOPATH=/private/tmp/cockroach20151218-16645-rwk1e1
COLOR_WHITE=\e[1;37m
HOMEBREW_OPTIMIZATION_LEVEL=Os
COLOR_LIGHT_GREEN=\e[1;32m
ACLOCAL_PATH=/usr/local/share/aclocal
HOMEBREW_CELLAR=/usr/local/Cellar
CMAKE_PREFIX_PATH=/usr/local
CC=clang
DISPLAY=/private/tmp/com.apple.launchd.7GcvpXNd6a/org.macosforge.xquartz:0
OLDPWD=/private/tmp/cockroach20151218-16645-rwk1e1
_=/usr/bin/env
```

File: build/make-brew-formula.sh [hunk 2]
```diff
@@ -29,11 +28,20 @@ line=$(grep -n '### GO RESOURCES ###' ${template} | cut -d ":" -f 1)
...
     echo
     head -n $((line-1)) ${template} | sed -e s/@VERSION@/${version}/g -e s/@REVISION@/${revision}/g
-    # homebrew-go-resources outputs some urls with "https///". We
-    # could potentially generate this section from the GLOCKFILE,
-    # though the dependencies there are more than what we need to
-    # build just the cockroach binary.
-    homebrew-go-resources . | sed 's,https///,https://,g'
+
+    for dep in $(build/depvers.sh); do
+        repo=$(echo ${dep} | cut -f1 -d:)
```
Comment (bdarnell) [inline build/make-brew-formula.sh]: `build/depvers.sh | while IFS=":" read repo rev; do`?

File: build/cockroach.rb (hunk 1 above)
Comment (bdarnell) [inline build/cockroach.rb]: Nothing rings any bells here. It sets `CC=clang` but that should be the default anyway. It does put its own directory at the beginning of `$PATH` so I suppose there could be something weird there, but I don't have any specific ideas.

**Comment by bdarnell** [comment]
LGTM

File: build/cockroach.rb (hunk 1 above)
Comment (petermattis) [inline build/cockroach.rb]: The culprit is `HOMEBREW_CCCFG`. In the shell this env var has the value `sa`. Within make the value is `Osa`. And somehow it is affecting the go build:

```
bash-3.2$ HOMEBREW_CCCFG=sa go build -v github.com/cockroachdb/c-snappy
github.com/cockroachdb/c-snappy
bash-3.2$ HOMEBREW_CCCFG=Osa go build -v github.com/cockroachdb/c-snappy
github.com/cockroachdb/c-snappy
# github.com/cockroachdb/c-snappy
cannot load DWARF output from $WORK/github.com/cockroachdb/c-snappy/_obj//_cgo_.o: decoding dwarf section info at offset 0x0: too short
```

Comment (petermattis) [inline build/cockroach.rb]:
```
    # The HOMEBREW_CCCFG ENV variable is used by the ENV/cc tool to control
    # compiler flag stripping. It consists of a string of characters which act
    # as flags. Some of these flags are mutually exclusive.
    #
    # O - Enables argument refurbishing. Only active under the
    #     make/bsdmake wrappers currently.
    # x - Enable C++11 mode.
    # g - Enable "-stdlib=libc++" for clang.
    # h - Enable "-stdlib=libstdc++" for clang.
    # K - Don't strip -arch <arch>, -m32, or -m64
    #
    # On 10.8 and newer, these flags will also be present:
    # s - apply fix for sed's Unicode support
    # a - apply fix for apr-1-config path
```

Comment (petermattis) [inline build/cockroach.rb]: Add this fixes the problem: `make build SKIP_BOOTSTRAP=1 HOMEBREW_CCCFG=${HOMEBREW_CCCFG}`

File: build/make-brew-formula.sh (hunk 2 above)
Comment (petermattis) [inline build/make-brew-formula.sh]: That's nicer. Done.

File: build/cockroach.rb (hunk 1 above)
Comment (petermattis) [inline build/cockroach.rb]: FYI, I'm going to fix this in a subsequent PR.

Comment (bdarnell) [inline build/cockroach.rb]: > FYI, I'm going to fix this in a subsequent PR.

SGTM.

The env var is getting modified by https://github.com/Homebrew/homebrew/blob/8c6777b2b8e847d5f2b447eae532fa8ecffa6467/Library/ENV/4.3/make and interpreted by https://github.com/Homebrew/homebrew/blob/8c6777b2b8e847d5f2b447eae532fa8ecffa6467/Library/ENV/4.3/cc. Looks like it strips out some debug-info-related flags which means that go can't get the metadata it needs. Bizarre that that's tied to whether it's running under `make` or not.

**Pull request merged**